notary-commons/src/main/kotlin/com/d3/commons/sidechain/iroha/util/impl/RobustIrohaQueryHelperImpl.kt

```diff
@@ -0,0 +1,128 @@
+/*
+ * Copyright D3 Ledger, Inc. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package com.d3.commons.sidechain.iroha.util.impl
+
+import com.d3.commons.sidechain.iroha.util.IrohaQueryHelper
+import com.github.kittinunf.result.Result
+import io.grpc.Status
+import io.grpc.StatusRuntimeException
+import mu.KLogging
+
+private const val TIMEOUT_MLS = 5_000L
+private const val MIN_TIMEOUT = TIMEOUT_MLS * 2
+
+/**
+ * Class that wraps [IrohaQueryHelperImpl]. Unlike [IrohaQueryHelperImpl] it tries to query Iroha multiple times if Iroha is off-line.
+ * @param irohaQueryHelper - [IrohaQueryHelperImpl] instance to wrap
+ * @param totalTimeoutMls - max time to retry in milliseconds.
+ * In fact it will be in range (totalTimeoutMls-TIMEOUT_MLS;totalTimeoutMls)
```

Comment:
Hard to comprehence. Please, add spaces `(totalTimeoutMls - TIMEOUT_MLS; totalTimeoutMls)`.